test/micro/org/openjdk/bench/java/lang/foreign/StrLenTest.java

```diff
@@ -154,17 +155,17 @@ public RingAllocator(MemorySession session) {
 
         @Override
         public MemorySegment allocate(long byteSize, long byteAlignment) {
-            MemorySegment res = current.allocate(byteSize, byteAlignment);
-            if (res.equals(MemorySegment.NULL)) {
-                // reached the end of the segment, start again
+            if (rem < byteSize) {
                 reset();
-                res = current.allocate(byteSize, byteAlignment);
             }
+            MemorySegment res = current.allocate(byteSize, byteAlignment);
+            rem -= segment.segmentOffset(res);
```

Comment:
I think this should be `rem = segment.byteSize() - segment.segmentOffset(res);` right? Otherwise more and more is subtracted on subsequent allocations.

Reply:
good catch!